app/inc/constants.php

```diff
@@ -42,6 +42,22 @@
     define('CACHE_TIME',  19200);
 }
 
+$api_root = APP_SCHEME . htmlspecialchars($_SERVER['HTTP_HOST'], ENT_QUOTES, 'UTF-8');
+
+// trying to find the root path of transvision
+$root_path = explode('transvision', htmlspecialchars($_SERVER['REQUEST_URI'], ENT_QUOTES, 'UTF-8'));
```

Comment:
My bad, I think in my previous comment I was confused with other projects were the webroot can be anything. On Transvision, we actually can’t have anything like `https://domain.tld/foo/bar/api/v1`. It will always be `https://domain.tld/api/v1/`.
So this should be enough:
`APP_SCHEME . htmlspecialchars($_SERVER['HTTP_HOST'], ENT_QUOTES, 'UTF-8') . '/api/v1/';` (probably what you did in the previous version)
However, HTTP_HOST is not necessarily defined, so we should at least check it’s defined. Not sure if gethostname would work with php integrated server, but might be worth trying https://secure.php.net/manual/en/function.gethostname.php